library/init/meta/interactive.lean

```diff
@@ -6,8 +6,11 @@ Authors: Leonardo de Moura
 prelude
 import init.meta.tactic init.meta.rewrite_tactic init.meta.simp_tactic
 import init.meta.smt.congruence_closure init.category.combinators
+import init.meta.lean_parser
 
 namespace interactive
+@[reducible] meta def parse {α : Type} (p : lean_parser α) : Type := α
```

Comment:
@kha I think the `has_quote` approach works well for parsing values (e.g., closed terms of type `nat` or `name`). It is strongly typed, and it is convenient to use (the conversion from value to the `pexpr` that represents it is done automatically by using `has_quote`). So, we could call it
```lean
 meta def parse_value {α : Type} (p : lean_parser α) : Type := α
```
This seems to work even if we have a `pexpr` which is not a closed term.
I agree this would cover many of the types we currently have.

We can keep `itactic` as is. I think this would cover everything we have right now. If we want to be on the safe side, we may also provide a weakly typed version that can be used to parse arbitrary expressions. Or, force the user to use keyword based parser extensions when they want to do crazy stuff.